### PR TITLE
feat(mysql): Add text search relevance to mysql

### PR DESCRIPTION
**Pull request by garrensmith**
Adds `visit_text_search_relevance` support to mysql and refactors the
original postgres only support to allow any connector to be able to
implement it.

### CODE REVIEW

File: src/visitor/mssql.rs [hunk 1]
```diff
@@ -620,6 +620,14 @@ impl<'a> Visitor<'a> for Mssql<'a> {
         unimplemented!("Full-text search is not yet supported on MSSQL")
     }
 
+    #[cfg(feature = "postgresql")]
```
Comment (dpetrick) [inline src/visitor/mssql.rs:623]: This looks off, why is there a postgres feature flag in mssql?

File: src/visitor/mysql.rs [hunk 2]
```diff
@@ -475,6 +475,18 @@ impl<'a> Visitor<'a> for Mysql<'a> {
         Ok(())
     }
 
+    #[cfg(any(feature = "postgresql", feature = "mysql"))]
```
Comment (dpetrick) [inline src/visitor/mysql.rs:478]: Do you really need these feature flags? The whole mysql struct is gated by the `mysql` feature flag already.

File: src/visitor/postgres.rs [hunk 3]
```diff
@@ -394,6 +394,31 @@ impl<'a> Visitor<'a> for Postgres<'a> {
 
         Ok(())
     }
+
+    #[cfg(any(feature = "postgresql", feature = "mysql"))]
```
Comment (dpetrick) [inline src/visitor/postgres.rs:398]: Same as the above comment.

File: src/visitor/mssql.rs (hunk 1 above)
Comment (garrensmith) [reply src/visitor/mssql.rs:623]: This is needed for when we have `features=all`

**Review by dpetrick — APPROVED** [review]

**Pull request merged**